src/commands/slash/database/Incident.ts

```diff
@@ -0,0 +1,98 @@
+import { SlashCommandBuilder, ChatInputCommandInteraction, GuildMemberRoleManager } from "discord.js";
+
+import { configuration } from "../../../server";
+import { Database } from "../../../utils/constants";
+import { Incident as IncidentModel } from "../../../models/DBModels";
+import { DatabaseCommand } from "./DatabaseCommand";
+
+export default new DatabaseCommand<IncidentModel>(Database.Containers.INCIDENTS, {
+    name: "incident",
+    description: "How many days since our last incident?",
+    help: "incident",
+    slashCommandDescription: getCommandDescription,
+    executeSlashCommand: slashCommandHandler,
+});
+
+enum SubCommands {
+    LAST = "last",
+    NEW = "new",
+}
+
+function getCommandDescription() {
+    const cmd = new SlashCommandBuilder()
+        .setName("incident")
+        .setDescription("check how many days have passed since our last `incident`");
+    cmd.addSubcommand((cmd) => cmd.setName("last").setDescription("days since last `incident`")).addSubcommand((cmd) =>
+        cmd
+            .setName("new")
+            .setDescription("indicates that a new incident has occurred")
+            .addStringOption((option) => option.setName("link").setDescription("message link to start of incident"))
+            .addStringOption((option) => option.setName("note").setDescription("what went down"))
+    );
+    return cmd;
+}
+
+async function slashCommandHandler(this: DatabaseCommand<IncidentModel>, interaction: ChatInputCommandInteraction) {
+    const cmd = interaction.options.getSubcommand(true);
+    if (cmd === SubCommands.LAST) {
+        //not a private response
+        await interaction.deferReply({ ephemeral: false });
+        // find the last incident
+        const incident = await this.connector.getLastItem();
+        if (!incident) {
+            // uh, we got a problem
+            interaction.followUp({
+                content: "No incidents yet. Hmm...",
+                ephemeral: true,
+            });
+            return;
+        }
+        // do some math
+        const timeSince = Date.now() - new Date(incident?.occurrence).getTime();
+        const daysSince = timeSince / (1000 * 60 * 60 * 24);
```

Comment:
Extract the numbers to a constant

Reply:
Instead, I've established a series of time-related utility functions.